src/CollisionEvent.cc

```diff
@@ -124,16 +124,18 @@ bool CollisionEvent(MonteCarlo* monteCarlo, MC_Particle &mc_particle, unsigned i
 
       randomNumber = rngSample(&mc_particle.random_number_seed);
       currentParticle->num_mean_free_paths = -1.0*log(randomNumber);
-      if (secondaryIndex > 0)
+      if( energy_angle_size > 1 )
```

Comment:
Please refactor as we discussed to eliminate frequently ignored copy of mc_particle